src/Opaleye/PGTypes.hs

```diff
@@ -38,6 +38,16 @@ instance C.PGNum PGInt8 where
 instance C.PGFractional PGFloat8 where
   pgFromRational = pgDouble . fromRational
 
+instance C.PGNum Int where pgFromInteger    = unsafeCoerceColumn . pgInt4 . fromIntegral
```

Comment:
On 64 bit this is would make insertion unsafe. However we decide on this instance an alternative would be to tell people to use Int32/Int64.